yaml/ver10/cloudintegration/yaml/cloudintegration.yaml

```diff
@@ -0,0 +1,149 @@
+
+components:
+  schemas:
+    Base64DERencodedASN1Value:
+      description: This schema is the equivalent of tas:Base64DERencodedASN1Value in the Security Service specification.
+      type: string
+      format: byte
+      examples: [ Base64DERencodedASN1Value ]
+
+    EndpointReferenceType:
+      description: This schema is the equivalent of wsa:EndpointReferenceType in the WA-Addressing 1.0 specification.
+      type: string
+      format: uri
+
+    AuthorizationServerConfiguration:
+      type: object
+      description: This schema is the equivalent of tas:AuthorizationServerConfigurationData
+      properties:
+        ServerURI:
+          description: Authorization server address
+          type: string
+        ClientID:
+          description: Client identifier issued by the authorization server
+          type: string
+        Scope:
+          description: The requested access scope
+          type: string
+      required:
+        - ServerURI
+        - ClientID
+
+    UplinkConfiguration:
+      type: object
+      properties:
+        RemoteAddress:
+          description: Uniform resource locator by which the remote client can be reached.
+          item:
+            $ref: '#/components/schemas/EndpointReferenceType'
+        UserLevel:
+          description: Authorization level that will be assigned to the uplink connection.
+          type: string
+          enum: [Administrator, Operator, User]
+        UplinkTrustAnchor:
+          items:
+            $ref: '#/components/schemas/Base64DERencodedASN1Value'
+          description: The certificate with the public key used to validate the TLS certificate of the uplink server.
+      required:
+        - RemoteAddress
+        - UserLevel
+
+    DeviceConfiguration:
+      type: object
+      properties:
+        ShareToken:
+          description: The share token, to prove permission to share the device.
+          item:
+            type: string
+            format: byte
+        EndpointReferences:
+          description: The endpoint to be notified to inform that the operation was completed. It may be more then one if the requesting cloud has redundant URIs to receive the notification.
+          items:
+            $ref: '#/components/schemas/EndpointReferenceType'
+        UplinkConfiguration:
+          description: Per-device uplink configuration
+          items:
+            $ref: '#/components/schemas/UplinkConfiguration'
+        AuthorizationServerConfiguration:
+          items:
+            $ref: '#/components/schemas/AuthorizationServerConfiguration'
+      required:
+        - ShareToken
+        - EndpointReferences
+        - UplinkConfiguration
+        - AuthorizationServerConfiguration
+
+    DeviceInformation:
+      type: object
+      properties:
+        ShareToken:
+          description: The share token, to know what device the public key is associated to.
+          $ref: '#/components/schemas/Base64DERencodedASN1Value'
+        DevicePublicKey:
```

Comment:
Right now nothing defines what kind of public keys the authorization server supports, nor what kind the device supports.
Thinking ahead where the algorithms for signing the tokens might change, we should probably have a way to signal what the server supports during the transfer call (and a corresponding error if the device cannot match any of them)